Framework/Algorithms/src/NormaliseToMonitor.cpp

```diff
@@ -283,30 +271,73 @@ void NormaliseToMonitor::exec() {
   }
 }
 
+/** Validates input properties.
+ *  @return A map of input properties as keys and (error) messages as values.
+ */
+std::map<std::string, std::string> NormaliseToMonitor::validateInputs() {
+  std::map<std::string, std::string> issues;
+  // Check where the monitor spectrum should come from
+  const Property *monSpecProp = getProperty("MonitorSpectrum");
+  const Property *monIDProp = getProperty("MonitorID");
+  MatrixWorkspace_const_sptr monWS = getProperty("MonitorWorkspace");
+  // something has to be set
+  if (monSpecProp->isDefault() && !monWS && monIDProp->isDefault()) {
+    const std::string mess("Either MonitorSpectrum, MonitorID or "
+                           "MonitorWorkspace has to be provided.");
+    issues["MonitorSpectrum"] = mess;
+    issues["MonitorID"] = mess;
+    issues["MonitorWorkspace"] = mess;
+  }
+
+  const double intMin = getProperty("IntegrationRangeMin");
+  const double intMax = getProperty("IntegrationRangeMax");
+  if (!isEmpty(intMin) && !isEmpty(intMax)) {
+    if (intMin > intMax) {
+      issues["IntegrationRangeMin"] =
+          "Integration minimum set to larger value than maximum.";
+    }
+  }
+
+  if (monWS) {
```

Comment:
There is a minor annoyance here - if a Monitor ID is set the other fields are disabled in the GUI. These checks are still done anyway, so I would add `!monIDProp->isDefault()` to avoid the user having to delete the monitor ID, clear the fields, and re-add the monitor ID.